### PR TITLE
bump postgres versions in CI and dev

### PR DESCRIPTION
**Pull request by thanodnl**
Upgrade postgres versions to:
 - 14.12
 -  15.7
 - 16.3

Depends on https://github.com/citusdata/the-process/pull/158

--

Checklist:
 - [x] update `image_suffix`

### CODE REVIEW

File: .github/workflows/build_and_test.yml
Comment (thanodnl) [inline .github/workflows/build_and_test.yml:35]: needs to be updated before this PR can land

Comment (hanefi) [reply .github/workflows/build_and_test.yml:35]:
```suggestion
      image_suffix: "-v13fd57c"
```

**Review by hanefi — APPROVED** [review]
Good to merge after we update image suffix.

**Pull request merged**
